Fix IIIF response headers and content-type detection

neilberkman opened the pull request:
Fix Content-Type headers in IIIF responses

The library was not setting Content-Type headers, causing them to fall back to application/octet-stream. This fix ensures proper content-type headers are set based on the requested output format (image/jpeg, image/png, etc).

- Fixes response header application from DataRequest callbacks  
- Adds automatic content-type detection based on output format
- Includes comprehensive tests for all send methods

neilberkman committed:
Fix IIIF response headers and content-type detection

- Apply response_headers from DataRequest to all send methods (send_buffered, send_stream, send_temporary_file)
- Auto-detect proper content-type based on output format instead of defaulting to application/octet-stream
- Add format_to_content_type mapping for common image formats (JPEG, PNG, WebP, TIFF, etc.)
- Preserve custom content-type headers when explicitly set in DataRequest.response_headers
- Add comprehensive tests for content-type detection across all send methods
- Add test plug for custom content-type scenarios

This fixes the issue where IIIF images were being served with incorrect
content-type headers causing browsers to download files instead of displaying them inline.

dersmon commented on lib/iiif_image_plug/v3.ex:
```diff
@@ -393,6 +393,9 @@ defmodule IIIFImagePlug.V3 do
   defp send_buffered(conn, %Image{} = image, format) do
     {:ok, buffer} = Image.write_to_buffer(image, ".#{format}")
 
+    # Use the requested format for content-type (since that's what we're delivering)
+    conn = put_resp_content_type_from_format(conn, format)
+
```
Maybe we could add these calls between line 329 and 330, before it branches off to the different response types?

dersmon reviewed (changes requested):
Thanks again! I had only one small suggestion.

neilberkman committed:
Move content-type setting to before response branching

As suggested in PR review, move the put_resp_content_type_from_format call
to occur before the cond statement that branches into different response
methods. This ensures consistent content-type handling across all response
paths (send_buffered, send_temporary_file, and send_stream).
